test/images/sample-device-plugin/sampledeviceplugin.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1beta1"
+	dm "k8s.io/kubernetes/pkg/kubelet/cm/devicemanager"
+)
+
+const (
+	resourceName = "fake.com/resource"
+)
+
+// stubAllocFunc will pass to stub device plugin
+func stubAllocFunc(r *pluginapi.AllocateRequest, devs map[string]pluginapi.Device) (*pluginapi.AllocateResponse, error) {
+	var responses pluginapi.AllocateResponse
+	for _, req := range r.ContainerRequests {
+		response := &pluginapi.ContainerAllocateResponse{}
+		for _, requestID := range req.DevicesIDs {
+			dev, ok := devs[requestID]
+			if !ok {
+				return nil, fmt.Errorf("invalid allocation request with non-existing device %s", requestID)
+			}
+
+			if dev.Health != pluginapi.Healthy {
+				return nil, fmt.Errorf("invalid allocation request with unhealthy device: %s", requestID)
+			}
+
+			// create fake device file
+			fpath := filepath.Join("/tmp", dev.ID)
+
+			// clean first
+			os.RemoveAll(fpath)
+			f, err := os.Create(fpath)
+			if err != nil && !os.IsExist(err) {
+				return nil, fmt.Errorf("failed to create fake device file: %s", err)
+			}
+
+			f.Close()
+
+			response.Mounts = append(response.Mounts, &pluginapi.Mount{
+				ContainerPath: fpath,
+				HostPath:      fpath,
+			})
+		}
+		responses.ContainerResponses = append(responses.ContainerResponses, response)
+	}
+
+	return &responses, nil
+}
+
+func main() {
+	devs := []*pluginapi.Device{
+		{ID: "Dev-1", Health: pluginapi.Healthy},
+		{ID: "Dev-2", Health: pluginapi.Healthy},
+	}
+
+	pluginSocksDir := os.Getenv("PLUGIN_SOCK_DIR")
+	fmt.Println("pluginSocksDir: ", pluginSocksDir)
+	if pluginSocksDir == "" {
```

Comment:
Should there be some retry logic (retry 3 times at 5s interval) or is it unnecessary?

Reply:
In the test we are verifying that kubelet is ready before starting plugin so seems kind of not adding much value.